### PR TITLE
Add HAProxy integration

### PR DESCRIPTION
**Pull request by echlebek**
This is an MVP for the HAProxy integration. At this time, there are no supported SumoLogic dashboards. However, the integration can be used to set up a working SumoLogic metrics collector.

Signed-off-by: Eric Chlebek <eric@sensu.io>

### CODE REVIEW

File: integrations/haproxy/haproxy-monitoring/sensu-integration.yaml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  class: community
+  provider: monitoring
+  display_name: HAProxy monitoring
+  short_description: Monitor HAProxy service health and performance
+  supported_platforms:
+    - darwin
+    - linux
+    - windows
+  tags:
+    - http
```
Comment (jspaleta) [inline integrations/haproxy/haproxy-monitoring/sensu-integration.yaml:17]:
```suggestion
    - service
    - http
```

File: integrations/haproxy/haproxy-monitoring/README.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+## Alerts
+
+<!-- List of all alerts generated by this integration. -->
+
+Currently, this integration only produces alerts when metrics collection fails.
+
+## Reference Documentation
+
+<!-- Please provide links to any relevant reference documentation to help users learn more and/or troubleshoot this integration. -->
+
```
Comment (jspaleta) [inline integrations/haproxy/haproxy-monitoring/README.md:139]: 1.  HAProxy management guide: [Statiistics and monitoring](https://cbonte.github.io/haproxy-dconv/2.4/management.html#9)

File: integrations/haproxy/haproxy-monitoring/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog][changelog] and this project adheres
+to [Semantic Versioning][semver].
+
+## Unreleased
```
Comment (jspaleta) [inline integrations/haproxy/haproxy-monitoring/CHANGELOG.md:8]: Ill clean this up after merge as I tag

**Comment by echlebek** [comment]
Feedback integrated!

**Review by jspaleta — APPROVED** [review]

**Pull request merged**